### PR TITLE
Mail

### PR DESCRIPTION
**Pull request by elijahkim**

### CODE REVIEW

File: web/services/password_resetter.ex [hunk 1]
```diff
@@ -0,0 +1,26 @@
+defmodule Looseleaf.PasswordResetter do
+  alias Looseleaf.{Email, Mailer, User}
+
+  def reset_password(email, repo) do
```
Comment (cjbell) [inline web/services/password_resetter.ex:4]: Any reason you're passing the repo in? are you doing that for all services like this?

File: web/controllers/password_reset_controller.ex [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  plug :put_layout, "home.html"
+
+  def new(conn, _params) do
+    conn
+    |> render("new.html")
+  end
+
+  def create(conn, %{"password_reset" => %{"email" => email}}) do
+    PasswordResetter.reset_password(email, Repo)
```
Comment (cjbell) [inline web/controllers/password_reset_controller.ex:13]: Nit but this service call might fail and return an `{:error, changeset}` – should you handle?

File: test/services/password_reseter_test.exs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  import Looseleaf.Factory
+  alias Looseleaf.Repo
+
+  describe "reset_password(user, repo)" do
+    test "it sets a token to edit a password" do
+      user = insert(:user)
+
+      {:ok, user, _email} = Looseleaf.PasswordResetter.reset_password(user.email, Repo)
+
+      assert user.password_reset_token != nil
```
Comment (cjbell) [inline test/services/password_reseter_test.exs]: You can just `assert user.password_reset_token` here or better possibly to use `refute user.password_reset_token == nil` if you're into that kind of thing

File: web/services/password_resetter.ex (hunk 1 above)
Comment (elijahkim) [reply web/services/password_resetter.ex:4]: yeah I think all of them. DI is cool.

File: web/controllers/password_reset_controller.ex (hunk 2 above)
Comment (elijahkim) [reply web/controllers/password_reset_controller.ex:13]: yep I should full feature isn't done. It's just getting big.

File: web/services/password_resetter.ex (hunk 1 above)
Comment (cjbell) [reply web/services/password_resetter.ex:4]: DI is cool, sometimes.

**Pull request merged**